#
# Backported from OpenBSD:
#
# Move C/R -> kbdint special case to after the defaults have been
# loaded, which makes ChallengeResponse default to yes again.  This
# was broken by the Match changes and not fixed properly subsequently.
# Found by okan at demirmen.com, ok djm@ "please do it" deraadt@
#
--- servconf.c.orig	2007-08-30 11:50:17.000000000 -0300
+++ servconf.c	2007-08-30 11:50:38.000000000 -0300
@@ -1387,8 +1387,4 @@
 	if (bad_options > 0)
 		fatal("%s: terminating, %d bad configuration options",
 		    filename, bad_options);
-
-	/* challenge-response is implemented via keyboard interactive */
-	if (options->challenge_response_authentication == 1)
-		options->kbd_interactive_authentication = 1;
 }
--- sshd.c.orig	2007-08-30 11:50:30.000000000 -0300
+++ sshd.c	2007-08-30 11:51:42.000000000 -0300
@@ -1421,6 +1421,10 @@
 	/* Fill in default values for those options not explicitly set. */
 	fill_default_server_options(&options);
 
+	/* challenge-response is implemented via keyboard interactive */
+	if (options.challenge_response_authentication)
+		options.kbd_interactive_authentication = 1;
+
 	/* set default channel AF */
 	channel_set_af(options.address_family);
 
